Rectangle/WindowCalculation/MoveUpDownCalculation.swift

```diff
@@ -35,16 +35,8 @@ class MoveUpDownCalculation: WindowCalculation, RepeatedExecutionsCalculation {
 
     }
     
-    func calculateFirstRect(_ params: RectCalculationParameters) -> RectResult {
-        return calculateGenericRect(params, fraction: 1 / 2.0)
-    }
-    
-    func calculateSecondRect(_ params: RectCalculationParameters) -> RectResult {
-        return calculateGenericRect(params, fraction: 2 / 3.0)
-    }
-    
-    func calculateThirdRect(_ params: RectCalculationParameters) -> RectResult {
-        return calculateGenericRect(params, fraction: 1 / 3.0)
+    func calculateFractionalRect(_ params: RectCalculationParameters, fraction: Float) -> RectResult {
+        return calculateGenericRect(params, fraction: fraction)
```

Comment:
This one was slightly frustrating, as `calculateFractionalRect(_ params: RectCalculationParameters, fraction: Float?)` isn’t a valid implementation of the protocol’s `calculateFractionalRect(_ params: RectCalculationParameters, fraction: Float)`, so I had to separate into `calculateFractionalRect` and `calculateGenericRect`.

Reply:
Right, that's not a big deal to me to have it call calculateGenericRect.